## **Scomposizione Titanic Dataset**

In [41]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

import warnings
warnings.filterwarnings("ignore")

class_names = {1: 'First Class', 2: 'Second Class', 3: 'Third Class'}

Import totale:

In [42]:
train = pd.read_csv("../Dataset/Titanic/train.csv")
test = pd.read_csv("../Dataset/Titanic/test.csv")
full = pd.concat([train, test])
full

PassengerId  Survived  Pclass  \
0              1       0.0       3   
1              2       1.0       1   
2              3       1.0       3   
3              4       1.0       1   
4              5       0.0       3   
..           ...       ...     ...   
413         1305       NaN       3   
414         1306       NaN       1   
415         1307       NaN       3   
416         1308       NaN       3   
417         1309       NaN       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
413                                 Spector, Mr. Woolf    male   NaN      0   
414                       Oliva y Ocana, Dona. Fermina  female  39.0      0   
415                       Saether, Mr. Simon Sivertsen    male  38.5      0   
416                                Ware, Mr. Frederick    male   NaN      0   
417                           Peter, Master. Michael J    male   NaN      1   

     Parch              Ticket      Fare Cabin Embarked  
0        0           A/5 21171    7.2500   NaN        S  
1        0            PC 17599   71.2833   C85        C  
2        0    STON/O2. 3101282    7.9250   NaN        S  
3        0              113803   53.1000  C123        S  
4        0              373450    8.0500   NaN        S  
..     ...                 ...       ...   ...      ...  
413      0           A.5. 3236    8.0500   NaN        S  
414      0            PC 17758  108.9000  C105        C  
415      0  SOTON/O.Q. 3101262    7.2500   NaN        S  
416      0              359309    8.0500   NaN        S  
417      1                2668   22.3583   NaN        C  

[1309 rows x 12 columns]

#### **1) Mapping Nomi - Passenger ID:**

In [43]:
ids = full[['PassengerId', 'Name']]
ids.columns = ['Passenger Id', 'Name Orig']
ids['Surname'] = ids['Name Orig'].apply(lambda x: x.split(", ")[0])
ids['Name'] = ids['Name Orig'].apply(lambda x: x.split(", ")[1])
ids.sort_values(by='Passenger Id', inplace=True)
ids.reset_index(drop=True, inplace=True)
ids

Passenger Id                                          Name Orig  \
0                1                            Braund, Mr. Owen Harris   
1                2  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2                3                             Heikkinen, Miss. Laina   
3                4       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4                5                           Allen, Mr. William Henry   
...            ...                                                ...   
1304          1305                                 Spector, Mr. Woolf   
1305          1306                       Oliva y Ocana, Dona. Fermina   
1306          1307                       Saether, Mr. Simon Sivertsen   
1307          1308                                Ware, Mr. Frederick   
1308          1309                           Peter, Master. Michael J   

            Surname                                        Name  
0            Braund                             Mr. Owen Harris  
1           Cumings  Mrs. John Bradley (Florence Briggs Thayer)  
2         Heikkinen                                 Miss. Laina  
3          Futrelle          Mrs. Jacques Heath (Lily May Peel)  
4             Allen                           Mr. William Henry  
...             ...                                         ...  
1304        Spector                                   Mr. Woolf  
1305  Oliva y Ocana                               Dona. Fermina  
1306        Saether                         Mr. Simon Sivertsen  
1307           Ware                               Mr. Frederick  
1308          Peter                           Master. Michael J  

[1309 rows x 4 columns]

In [44]:
ids.drop(columns='Name Orig', inplace=True)
ids

Passenger Id        Surname                                        Name
0                1         Braund                             Mr. Owen Harris
1                2        Cumings  Mrs. John Bradley (Florence Briggs Thayer)
2                3      Heikkinen                                 Miss. Laina
3                4       Futrelle          Mrs. Jacques Heath (Lily May Peel)
4                5          Allen                           Mr. William Henry
...            ...            ...                                         ...
1304          1305        Spector                                   Mr. Woolf
1305          1306  Oliva y Ocana                               Dona. Fermina
1306          1307        Saether                         Mr. Simon Sivertsen
1307          1308           Ware                               Mr. Frederick
1308          1309          Peter                           Master. Michael J

[1309 rows x 3 columns]

In [45]:
ids.to_csv("../out/Titanic/passenger_id.csv", index=False)

#### **2) Anagrafe passeggeri**:

In [46]:
registry = full[['Name', 'Sex', 'Age', 'Pclass', 'Embarked']]
registry.sort_values(by='Name', inplace=True)
registry.rename(columns={'Name': 'Name Orig', 'Pclass': 'Pclass Orig'}, inplace=True)
registry.reset_index(drop=True, inplace=True)
registry

Name Orig     Sex   Age  \
0                               Abbing, Mr. Anthony    male  42.0   
1                     Abbott, Master. Eugene Joseph    male  13.0   
2                       Abbott, Mr. Rossmore Edward    male  16.0   
3                  Abbott, Mrs. Stanton (Rosa Hunt)  female  35.0   
4                       Abelseth, Miss. Karen Marie  female  16.0   
...                                             ...     ...   ...   
1304  del Carlo, Mrs. Sebastiano (Argenia Genovesi)  female  24.0   
1305            van Billiard, Master. James William    male   NaN   
1306              van Billiard, Master. Walter John    male  11.5   
1307                van Billiard, Mr. Austin Blyler    male  40.5   
1308                    van Melkebeke, Mr. Philemon    male   NaN   

      Pclass Orig Embarked  
0               3        S  
1               3        S  
2               3        S  
3               3        S  
4               3        S  
...           ...      ...  
1304            2        C  
1305            3        S  
1306            3        S  
1307            3        S  
1308            3        S  

[1309 rows x 5 columns]

In [47]:
registry['Pclass'] = [class_names[registry['Pclass Orig'].iloc[i]] for i in range(len(registry.index))]
registry['Surname'] = registry['Name Orig'].apply(lambda x: x.split(", ")[0])
registry['Name'] = registry['Name Orig'].apply(lambda x: x.split(", ")[1])
registry['Sex'] = np.where(registry['Sex'] == 'male', 'Male', 'Female')
registry

Name Orig     Sex   Age  \
0                               Abbing, Mr. Anthony    Male  42.0   
1                     Abbott, Master. Eugene Joseph    Male  13.0   
2                       Abbott, Mr. Rossmore Edward    Male  16.0   
3                  Abbott, Mrs. Stanton (Rosa Hunt)  Female  35.0   
4                       Abelseth, Miss. Karen Marie  Female  16.0   
...                                             ...     ...   ...   
1304  del Carlo, Mrs. Sebastiano (Argenia Genovesi)  Female  24.0   
1305            van Billiard, Master. James William    Male   NaN   
1306              van Billiard, Master. Walter John    Male  11.5   
1307                van Billiard, Mr. Austin Blyler    Male  40.5   
1308                    van Melkebeke, Mr. Philemon    Male   NaN   

      Pclass Orig Embarked        Pclass        Surname  \
0               3        S   Third Class         Abbing   
1               3        S   Third Class         Abbott   
2               3        S   Third Class         Abbott   
3               3        S   Third Class         Abbott   
4               3        S   Third Class       Abelseth   
...           ...      ...           ...            ...   
1304            2        C  Second Class      del Carlo   
1305            3        S   Third Class   van Billiard   
1306            3        S   Third Class   van Billiard   
1307            3        S   Third Class   van Billiard   
1308            3        S   Third Class  van Melkebeke   

                                    Name  
0                            Mr. Anthony  
1                  Master. Eugene Joseph  
2                    Mr. Rossmore Edward  
3               Mrs. Stanton (Rosa Hunt)  
4                      Miss. Karen Marie  
...                                  ...  
1304  Mrs. Sebastiano (Argenia Genovesi)  
1305               Master. James William  
1306                 Master. Walter John  
1307                   Mr. Austin Blyler  
1308                        Mr. Philemon  

[1309 rows x 8 columns]

In [48]:
registry_ok = pd.DataFrame({'Surname': registry['Surname'], 'Name': registry['Name'], 'Sex': registry['Sex'], 'Age': registry['Age'], 'Pclass': registry['Pclass'], 'Embarked': registry['Embarked']})
registry_ok

Surname                                Name     Sex   Age  \
0            Abbing                         Mr. Anthony    Male  42.0   
1            Abbott               Master. Eugene Joseph    Male  13.0   
2            Abbott                 Mr. Rossmore Edward    Male  16.0   
3            Abbott            Mrs. Stanton (Rosa Hunt)  Female  35.0   
4          Abelseth                   Miss. Karen Marie  Female  16.0   
...             ...                                 ...     ...   ...   
1304      del Carlo  Mrs. Sebastiano (Argenia Genovesi)  Female  24.0   
1305   van Billiard               Master. James William    Male   NaN   
1306   van Billiard                 Master. Walter John    Male  11.5   
1307   van Billiard                   Mr. Austin Blyler    Male  40.5   
1308  van Melkebeke                        Mr. Philemon    Male   NaN   

            Pclass Embarked  
0      Third Class        S  
1      Third Class        S  
2      Third Class        S  
3      Third Class        S  
4      Third Class        S  
...            ...      ...  
1304  Second Class        C  
1305   Third Class        S  
1306   Third Class        S  
1307   Third Class        S  
1308   Third Class        S  

[1309 rows x 6 columns]

In [49]:
registry_ok.to_csv("../out/Titanic/registry.csv", index=False)

#### **3) Biglietteria**

In [92]:
tickets = full[['Name', 'Ticket', 'Fare', 'Cabin', 'Pclass']]
tickets.reset_index(drop=True, inplace=True)
tickets

Name              Ticket  \
0                               Braund, Mr. Owen Harris           A/5 21171   
1     Cumings, Mrs. John Bradley (Florence Briggs Th...            PC 17599   
2                                Heikkinen, Miss. Laina    STON/O2. 3101282   
3          Futrelle, Mrs. Jacques Heath (Lily May Peel)              113803   
4                              Allen, Mr. William Henry              373450   
...                                                 ...                 ...   
1304                                 Spector, Mr. Woolf           A.5. 3236   
1305                       Oliva y Ocana, Dona. Fermina            PC 17758   
1306                       Saether, Mr. Simon Sivertsen  SOTON/O.Q. 3101262   
1307                                Ware, Mr. Frederick              359309   
1308                           Peter, Master. Michael J                2668   

          Fare Cabin  Pclass  
0       7.2500   NaN       3  
1      71.2833   C85       1  
2       7.9250   NaN       3  
3      53.1000  C123       1  
4       8.0500   NaN       3  
...        ...   ...     ...  
1304    8.0500   NaN       3  
1305  108.9000  C105       1  
1306    7.2500   NaN       3  
1307    8.0500   NaN       3  
1308   22.3583   NaN       3  

[1309 rows x 5 columns]

In [93]:
print(len(tickets['Ticket'].unique().tolist()))

929


Estrazione statistiche targets:

In [94]:
targets = tickets[['Pclass', 'Fare']].groupby(by='Pclass', as_index=False).mean()
targets['Fare'] = targets['Fare'].round(4)
targets

Pclass     Fare
0       1  87.5090
1       2  21.1792
2       3  13.3029

In [95]:
fares_map = {i: targets['Fare'].loc[i-1] for i in targets['Pclass']}
fares_map

{1: 87.509, 2: 21.1792, 3: 13.3029}

Aggiunta persone esterne:

In [96]:
real_extras = ['Hershey, Mr. Milton',
                'Morgan, Mr. J. Pierpont',
                'Marconi, Mr. Guglielmo',
                'Frick, Mr. Henry Clay',
                'Vanderbilt, Mr. Alfred Gwynne',
                'Dreiser, Mr. Theodore',
                'Mott, Mr. John',
                'Penney, Mr. James Cash',
                'Seiberling, Mr. Frank',
                'Dix, Mr. John Alden',
                'Adams, Mr. Henry']

In [97]:
fakes = pd.read_csv("../Dataset/Titanic/nomi_fake.csv", sep=';')
fakes.head()

Surname, Name
0        Sato, Mr. Hakaru
1  Takahashi, Mr. Haruaki
2    Tanaka, Mr. Haruhisa
3  Watanabe, Mr. Haruichi
4         Ito, Mr. Haruki

Add extras:

In [98]:
import random
ls_extras = real_extras + fakes['Surname, Name'].tolist()
ls_names = []
ls_tickets = []
ls_pclasses = []
ls_fares = []

for extra in ls_extras:
    ls_names.append(extra)
    ls_tickets.append(f"ET {random.randint(10000000, 99999999)}")
    pclass = 1 if extra in real_extras else random.randint(1, 3)
    ls_pclasses.append(pclass)
    ls_fares.append(fares_map[pclass])

extras = pd.DataFrame({'Name': ls_names, 'Ticket': ls_tickets, 'Fare': ls_fares, 'Pclass': ls_pclasses})
extras


Name       Ticket     Fare  Pclass
0             Hershey, Mr. Milton  ET 62704764  87.5090       1
1         Morgan, Mr. J. Pierpont  ET 78337573  87.5090       1
2          Marconi, Mr. Guglielmo  ET 45405267  87.5090       1
3           Frick, Mr. Henry Clay  ET 58359358  87.5090       1
4   Vanderbilt, Mr. Alfred Gwynne  ET 49941854  87.5090       1
5           Dreiser, Mr. Theodore  ET 17595661  87.5090       1
6                  Mott, Mr. John  ET 52900208  87.5090       1
7          Penney, Mr. James Cash  ET 65927369  87.5090       1
8           Seiberling, Mr. Frank  ET 53934529  87.5090       1
9             Dix, Mr. John Alden  ET 35173829  87.5090       1
10               Adams, Mr. Henry  ET 52419606  87.5090       1
11               Sato, Mr. Hakaru  ET 72017768  21.1792       2
12         Takahashi, Mr. Haruaki  ET 37575086  87.5090       1
13           Tanaka, Mr. Haruhisa  ET 44501790  21.1792       2
14         Watanabe, Mr. Haruichi  ET 69078213  87.5090       1
15                Ito, Mr. Haruki  ET 72560325  87.5090       1
16        Nakamura, Mr. Harukichi  ET 54596418  21.1792       2
17       Kobayashi, Mr. Harumitsu  ET 95963317  21.1792       2
18         Yamamoto, Mr. Harunobu  ET 87273506  87.5090       1
19             Kato, Mr. Harunori  ET 63943497  87.5090       1
20             Yoshida, Mr. Haruo  ET 78012126  87.5090       1
21           Yamada, Mr. Harutaka  ET 16214071  87.5090       1
22             Sasaki, Mr. Haruto  ET 48075937  87.5090       1
23          Yamaguchi, Mrs. Kanna  ET 30910054  21.1792       2
24         Matsumoto, Mrs. Kanoko  ET 99154489  87.5090       1
25              Inoue, Mrs. Kanon  ET 31095321  13.3029       3
26             Kimura, Mrs. Kaori  ET 75358163  21.1792       2
27          Shimizu, Mrs. Kaoruko  ET 29918194  13.3029       3
28           Hayashi, Mrs. Kasumi  ET 43922330  21.1792       2
29            Saito, Mrs. Katsuko  ET 67450395  13.3029       3
30            Saito, Mrs. Katsura  ET 87504713  13.3029       3
31           Yamazaki, Mrs. Kawai  ET 56791974  21.1792       2
32            Nakajima, Mrs. Kaya  ET 36719611  13.3029       3
33         Nakashima, Mrs. Kayoko  ET 17347684  87.5090       1
34               Mori, Mrs. Kazue  ET 73311628  13.3029       3
35                Abe, Mrs. Keiki  ET 46392478  13.3029       3
36               Ikeda, Mrs. Kiho  ET 64042975  21.1792       2
37           Hashimoto, Mrs. Kiko  ET 88511489  87.5090       1
38            Ishikawa, Mrs. Kiku  ET 42735721  21.1792       2
39          Yamashita, Mrs. Kikue  ET 57608143  21.1792       2
40             Ogawa, Mrs. Kikuko  ET 91791913  21.1792       2
41               Ishii, Mrs. Kimi  ET 56685214  13.3029       3
42          Hasegawa, Mrs. Kimiko  ET 66254713  87.5090       1
43              Goto, Mrs. Kinuko  ET 58066189  87.5090       1
44               Okada, Mrs. Kira  ET 57397746  13.3029       3
45             Kondo, Mrs. Kirari  ET 15503024  21.1792       2
46             Maeda, Mrs. Kiriko  ET 30772902  87.5090       1
47            Fujita, Mrs. Kiyoko  ET 59156142  13.3029       3
48              Endo, Mrs. Koharu  ET 62416420  21.1792       2
49              Aoki, Mrs. Komako  ET 99476280  13.3029       3
50          Sakamoto, Mrs. Konomi  ET 12613632  87.5090       1
51          Murakami, Mrs. Koruri  ET 58255839  13.3029       3
52                Ota, Mrs. Kotoe  ET 39560002  13.3029       3
53               Kaneko, Mrs. Kou  ET 36710643  13.3029       3
54              Fujii, Mrs. Kozue  ET 77201809  21.1792       2

In [100]:
tickets = pd.concat([tickets, extras])
tickets

Name            Ticket  \
0                             Braund, Mr. Owen Harris         A/5 21171   
1   Cumings, Mrs. John Bradley (Florence Briggs Th...          PC 17599   
2                              Heikkinen, Miss. Laina  STON/O2. 3101282   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)            113803   
4                            Allen, Mr. William Henry            373450   
..                                                ...               ...   
50                              Sakamoto, Mrs. Konomi       ET 12613632   
51                              Murakami, Mrs. Koruri       ET 58255839   
52                                    Ota, Mrs. Kotoe       ET 39560002   
53                                   Kaneko, Mrs. Kou       ET 36710643   
54                                  Fujii, Mrs. Kozue       ET 77201809   

       Fare Cabin  Pclass  
0    7.2500   NaN       3  
1   71.2833   C85       1  
2    7.9250   NaN       3  
3   53.1000  C123       1  
4    8.0500   NaN       3  
..      ...   ...     ...  
50  87.5090   NaN       1  
51  13.3029   NaN       3  
52  13.3029   NaN       3  
53  13.3029   NaN       3  
54  21.1792   NaN       2  

[1364 rows x 5 columns]

Prepare for export:

In [101]:
tickets.sort_values(by='Ticket', inplace=True)
tickets['Ticket Class'] = [class_names[tickets['Pclass'].iloc[i]] for i in range(len(tickets.index))]
tickets.rename(columns={'Name': 'Name Orig'}, inplace=True)
tickets['Surname'] = tickets['Name Orig'].apply(lambda x: x.split(", ")[0])
tickets['Name'] = tickets['Name Orig'].apply(lambda x: x.split(", ")[1])
tickets.reset_index(drop=True, inplace=True)
tickets

Name Orig       Ticket    Fare  \
0                                  Cherry, Miss. Gladys       110152  86.500   
1     Rothes, the Countess. of (Lucy Noel Martha Dye...       110152  86.500   
2                                 Maioni, Miss. Roberta       110152  86.500   
3                                   Taussig, Miss. Ruth       110413  79.650   
4                Taussig, Mrs. Emil (Tillie Mandelbaum)       110413  79.650   
...                                                 ...          ...     ...   
1359                        Chaffee, Mr. Herbert Fuller  W.E.P. 5734  61.175   
1360  Chaffee, Mrs. Herbert Fuller (Carrie Constance...  W.E.P. 5734  61.175   
1361                                 Harris, Mr. Walter    W/C 14208  10.500   
1362                            Crosby, Miss. Harriet R    WE/P 5735  71.000   
1363                       Crosby, Capt. Edward Gifford    WE/P 5735  71.000   

     Cabin  Pclass  Ticket Class  Surname  \
0      B77       1   First Class   Cherry   
1      B77       1   First Class   Rothes   
2      B79       1   First Class   Maioni   
3      E68       1   First Class  Taussig   
4      E67       1   First Class  Taussig   
...    ...     ...           ...      ...   
1359   E31       1   First Class  Chaffee   
1360   E31       1   First Class  Chaffee   
1361   NaN       2  Second Class   Harris   
1362   B22       1   First Class   Crosby   
1363   B22       1   First Class   Crosby   

                                                  Name  
0                                         Miss. Gladys  
1     the Countess. of (Lucy Noel Martha Dyer-Edwards)  
2                                        Miss. Roberta  
3                                           Miss. Ruth  
4                        Mrs. Emil (Tillie Mandelbaum)  
...                                                ...  
1359                                Mr. Herbert Fuller  
1360    Mrs. Herbert Fuller (Carrie Constance Toogood)  
1361                                        Mr. Walter  
1362                                   Miss. Harriet R  
1363                              Capt. Edward Gifford  

[1364 rows x 8 columns]

In [102]:
tickets_ok = pd.DataFrame({'Surname': tickets['Surname'], 'Name': tickets['Name'], 'Ticket': tickets['Ticket'], 'Fare': tickets['Fare'], 'Cabin': tickets['Cabin'], 'Ticket Class': tickets['Ticket Class']})
tickets_ok

Surname                                              Name       Ticket  \
0      Cherry                                      Miss. Gladys       110152   
1      Rothes  the Countess. of (Lucy Noel Martha Dyer-Edwards)       110152   
2      Maioni                                     Miss. Roberta       110152   
3     Taussig                                        Miss. Ruth       110413   
4     Taussig                     Mrs. Emil (Tillie Mandelbaum)       110413   
...       ...                                               ...          ...   
1359  Chaffee                                Mr. Herbert Fuller  W.E.P. 5734   
1360  Chaffee    Mrs. Herbert Fuller (Carrie Constance Toogood)  W.E.P. 5734   
1361   Harris                                        Mr. Walter    W/C 14208   
1362   Crosby                                   Miss. Harriet R    WE/P 5735   
1363   Crosby                              Capt. Edward Gifford    WE/P 5735   

        Fare Cabin  Ticket Class  
0     86.500   B77   First Class  
1     86.500   B77   First Class  
2     86.500   B79   First Class  
3     79.650   E68   First Class  
4     79.650   E67   First Class  
...      ...   ...           ...  
1359  61.175   E31   First Class  
1360  61.175   E31   First Class  
1361  10.500   NaN  Second Class  
1362  71.000   B22   First Class  
1363  71.000   B22   First Class  

[1364 rows x 6 columns]

In [103]:
tickets_ok.to_csv("../out/Titanic/tickets.csv", index=False)

#### **4) Famiglie**

In [105]:
families = full[['Name', 'SibSp', 'Parch']].loc[(full['SibSp'] > 0) | (full['Parch'] > 0)]
families.sort_values(by='Name', inplace=True)
families.rename(columns={'Name': 'Name Orig', 'SibSp': 'Sib Sp', 'Parch': 'Par Ch'}, inplace=True)
families['Surname'] = families['Name Orig'].apply(lambda x: x.split(", ")[0])
families['Name'] = families['Name Orig'].apply(lambda x: x.split(", ")[1])
families.reset_index(drop=True, inplace=True)
families

Name Orig  Sib Sp  Par Ch  \
0                    Abbott, Master. Eugene Joseph       0       2   
1                      Abbott, Mr. Rossmore Edward       1       1   
2                 Abbott, Mrs. Stanton (Rosa Hunt)       1       1   
3                              Abelson, Mr. Samuel       1       0   
4            Abelson, Mrs. Samuel (Hannah Wizosky)       1       0   
..                                             ...     ...     ...   
514                      del Carlo, Mr. Sebastiano       1       0   
515  del Carlo, Mrs. Sebastiano (Argenia Genovesi)       1       0   
516            van Billiard, Master. James William       1       1   
517              van Billiard, Master. Walter John       1       1   
518                van Billiard, Mr. Austin Blyler       0       2   

          Surname                                Name  
0          Abbott               Master. Eugene Joseph  
1          Abbott                 Mr. Rossmore Edward  
2          Abbott            Mrs. Stanton (Rosa Hunt)  
3         Abelson                          Mr. Samuel  
4         Abelson        Mrs. Samuel (Hannah Wizosky)  
..            ...                                 ...  
514     del Carlo                      Mr. Sebastiano  
515     del Carlo  Mrs. Sebastiano (Argenia Genovesi)  
516  van Billiard               Master. James William  
517  van Billiard                 Master. Walter John  
518  van Billiard                   Mr. Austin Blyler  

[519 rows x 5 columns]

In [106]:
print(len(families['Surname'].unique().tolist()))

209


In [107]:
families_ok = families[['Surname', 'Name', 'Sib Sp', 'Par Ch']]
families_ok

Surname                                Name  Sib Sp  Par Ch
0          Abbott               Master. Eugene Joseph       0       2
1          Abbott                 Mr. Rossmore Edward       1       1
2          Abbott            Mrs. Stanton (Rosa Hunt)       1       1
3         Abelson                          Mr. Samuel       1       0
4         Abelson        Mrs. Samuel (Hannah Wizosky)       1       0
..            ...                                 ...     ...     ...
514     del Carlo                      Mr. Sebastiano       1       0
515     del Carlo  Mrs. Sebastiano (Argenia Genovesi)       1       0
516  van Billiard               Master. James William       1       1
517  van Billiard                 Master. Walter John       1       1
518  van Billiard                   Mr. Austin Blyler       0       2

[519 rows x 4 columns]

In [108]:
families_ok.to_csv("../out/Titanic/families.csv", index=False)

#### **5) Sopravvissuti**

In [60]:
survived = full[['Name', 'Survived']].loc[full['Survived'] == 1]
survived.rename(columns={'Name': 'Name Orig'}, inplace=True)
survived['Surname'] = survived['Name Orig'].apply(lambda x: x.split(", ")[0])
survived['Name'] = survived['Name Orig'].apply(lambda x: x.split(", ")[1])
survived.reset_index(drop=True, inplace=True)
survived

Name Orig  Survived    Surname  \
0    Cumings, Mrs. John Bradley (Florence Briggs Th...       1.0    Cumings   
1                               Heikkinen, Miss. Laina       1.0  Heikkinen   
2         Futrelle, Mrs. Jacques Heath (Lily May Peel)       1.0   Futrelle   
3    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)       1.0    Johnson   
4                  Nasser, Mrs. Nicholas (Adele Achem)       1.0     Nasser   
..                                                 ...       ...        ...   
337                   Najib, Miss. Adele Kiamie "Jane"       1.0      Najib   
338      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)       1.0     Potter   
339       Shelley, Mrs. William (Imanita Parrish Hall)       1.0    Shelley   
340                       Graham, Miss. Margaret Edith       1.0     Graham   
341                              Behr, Mr. Karl Howell       1.0       Behr   

                                           Name  
0    Mrs. John Bradley (Florence Briggs Thayer)  
1                                   Miss. Laina  
2            Mrs. Jacques Heath (Lily May Peel)  
3      Mrs. Oscar W (Elisabeth Vilhelmina Berg)  
4                   Mrs. Nicholas (Adele Achem)  
..                                          ...  
337                   Miss. Adele Kiamie "Jane"  
338       Mrs. Thomas Jr (Lily Alexenia Wilson)  
339         Mrs. William (Imanita Parrish Hall)  
340                        Miss. Margaret Edith  
341                             Mr. Karl Howell  

[342 rows x 4 columns]

In [61]:
survived_ok = pd.DataFrame({'Surname': survived['Surname'], 'Name': survived['Name'], 'Survived': survived['Survived']})
survived_ok

Surname                                        Name  Survived
0      Cumings  Mrs. John Bradley (Florence Briggs Thayer)       1.0
1    Heikkinen                                 Miss. Laina       1.0
2     Futrelle          Mrs. Jacques Heath (Lily May Peel)       1.0
3      Johnson    Mrs. Oscar W (Elisabeth Vilhelmina Berg)       1.0
4       Nasser                 Mrs. Nicholas (Adele Achem)       1.0
..         ...                                         ...       ...
337      Najib                   Miss. Adele Kiamie "Jane"       1.0
338     Potter       Mrs. Thomas Jr (Lily Alexenia Wilson)       1.0
339    Shelley         Mrs. William (Imanita Parrish Hall)       1.0
340     Graham                        Miss. Margaret Edith       1.0
341       Behr                             Mr. Karl Howell       1.0

[342 rows x 3 columns]

In [62]:
survived_ok.to_csv("../out/Titanic/survived.csv", index=False)

TRAIN / TEST SPLIT:
* Train set: IDs da 1 a 891
* Test set: IDs da 892 a 1309
* Features: Pclass, sex, age, sib-sp, parch, fare

In [51]:
#gendsub['Survived'] = np.where(gendsub['Survived Bool']==1, 'Yes', 'No')
#gendsub.to_csv("../out/Titanic/casualties_full.csv", index=False)